### ISSUE TITLE
Sometimes measures distance to nearest node (on street), not nearest point / line

### ISSUE THREAD

**Issue by adam-lynch** [adam-lynch/back-of-your-hand#6]
I got an email mentioning this. There is code there to find the closest line / point on the line, so I'm not sure what's happening. Needs more investigation.

This is an example the user gave:

![Screenshot_20210102-020948](https://user-images.githubusercontent.com/1427241/103855313-55dad800-50aa-11eb-8a42-6602c8c11c9a.png)

The dashed distance line should be perpendicular to the street.

The street can be found in OpenStreetMap / Overpass Turbo using the street name ("Exhibition Street") and the center coordinates from the URL (-37.80329, 144.96576)

**Comment by adam-lynch** [comment]
https://news.ycombinator.com/item?id=30738618#30739062

**Comment by adam-lynch** [comment]
https://news.ycombinator.com/item?id=30742093

**Comment by adam-lynch** [comment]
I should set up unit tests for this. It could be headless / UI-less but I think Cypress tests that assert the outcomes *and* draws it visually on a map would be really nice.

**Comment by adam-lynch** [comment]
https://leafletjs.com/SlavaUkraini/reference.html#lineutil-simplify might be helpful

**Comment by adam-lynch** [comment]
I had to rewrite this completely but I'm very satisfied with how it works now.

I did try and set up tests with Cypress but ran into too many issues to make it worth it (right now anyway);

- Leaflet thought the device was touch/touch-only so things like dragging with a mouse broke.
- To fix the above (or attempt to), I loaded leaflet in the fixture HTML page. This meant I had to use `window.L` rather than  importing leaflet which is awkward in itself but also any source files that import leaflet would have to be modified to accept a leaflet instance parameter.
- I ran into a bundling issue with some turf utilities but that's probably my own fault.
